### DISCUSSION TITLE
Can Teleport handle server logins and request MFA when not connecting trough proxy?

### DISCUSSION THREAD

**Discussion by SnowManl** [gravitational/teleport#28656] [Q&A]
Can Teleport handle server logins and request MFA when not connecting trough proxy?
Because when I am accessing my servers trough teleport proxy and I ssh I get MFA request can I make telport take care of SSO when logging in in servers so it would also request MFA when logging in?
if there is a way how can I do it?

**Comment by webvictim** [comment]
Your question isn't completely clear to me. The Teleport auth server is the part which handles login/MFA - this is linked to the proxy, so when connecting via the proxy you are prompted to sign in and provide MFA etc.

This step is mandatory to be able to connect to servers via Teleport - it's generally not possible to connect directly to servers.

**Reply by SnowManl** [reply]
Sorry for not being clear basically what I am wondering if it is possible to make Teleport handle servers logins like for example connecting trough solar putty it would not only ask for password when logging in to the server but ask for mfa too via teleport.
Is it possible?

**Reply by webvictim** [reply]
I understand. As long as your client is running `tsh proxy ssh` under the hood (similar to the output of `tsh config`) then yes, it will prompt you to log in and use MFA.

**Reply by SnowManl** [reply]
Is there documentation how to do this somewhere?

**Reply by webvictim** [reply]
For Solar Putty specificially, there is not. But broadly, if your Teleport proxy lives at `teleport.example.com:443`, when you log into your cluster with `tsh login --proxy=teleport.example.com:443` and then run `tsh config`, you'll see the output that would be needed to configure an SSH client correctly.

The proxy command specifically will be something like this:

```tsh proxy ssh --proxy=teleport.example.com:443 --cluster teleport.example.com %r@%h:%p```

You'll need to modify/translate this for whatever other client you're using.

**Reply by SnowManl** [reply]
But will this prompt MFA logins when I try to ssh to any server from any provider to a server that belongs to the cluster?

**Reply by webvictim** [reply]
Yes.

**Reply by SnowManl** [reply]
Can i somehow edit and configure PAM for teleport to take care of it?

**Reply by webvictim** [reply]
No, the PAM stack on the machine only takes effect once you've already authenticated to it. Teleport doesn't provide a way for you to invoke its MFA authentication flow directly from a machine - you need to connect via the Teleport proxy and complete the MFA challenge _before_ connecting to the machine.
